test_verbosefilter.py

```diff
@@ -0,0 +1,108 @@
+# -*- python -*-
+
+"""
+Unit tests for VerboseFilter, a logging helper in adept_openstack.py
+"""
+
+import logging
+import sys
+from mock import Mock
+from unittest2 import TestCase, main
+
+sys.modules['virtualenv'] = Mock()      # Needed for Ed's Fedora system
+
+sys.path.insert(0, 'kommandir/bin')
+
+
+class MyHandler(logging.Handler):
+    """
+    Output handler for logging messages. No output; just push onto list.
+    """
+    def __init__(self, message_list):
+        logging.Handler.__init__(self)
+        self.message_list = message_list
+
+    def emit(self, record):
+        self.message_list.append(record.msg)
+
+
+class TestVerboseFilter(TestCase):
+    """
+    General test class. Defines only one test function, used by
+    test generator to create tests for all combinations of levels
+    and verbose flags.
+    """
+
+    def setUp(self):
+        import adept_openstack  # pylint: disable=E0401
+        self.a_o = adept_openstack
+        self.logger = logging.getLogger()
+        self.message_list = []
+
+    def _test_basic(self, level, verbose, expect):
+        # Grrr. The Logger thing is a global, and there doesn't seem to be
+        # a way to get a new one each time. For addHandler() and addFilter()
+        # to work, we need to delve into internals.
+        self.logger.handlers = []
+        self.logger.filters = []
+
+        # Now we can set up our desired handler and filter
+        self.logger.setLevel(level)
+        self.message_list = []
+        self.logger.addHandler(MyHandler(self.message_list))
+        self.logger.addFilter(self.a_o.VerboseFilter(level, verbose))
+
+        # Always log the same messages...     pylint: disable=C0326
+        logging.debug( "this is DEBUG")
+        logging.info(  "this is INFO")
+        logging.info( ">this is VERBOSE-INFO")
+        logging.warn(  "this is WARN")
+        logging.error( "this is ERROR")
+
+        # ...the difference is what we expect to see.
+        expect_list = []
+        for level_name in ['DEBUG', 'INFO', 'VERBOSE-INFO', 'WARN', 'ERROR']:
+            if level_name[0] in expect:
+                expect_list.append('this is {}'.format(level_name))
+
+        self.assertEqual(self.message_list, expect_list)
+
+def test_generator(test_info):
+    """
+    Generate a test for this combination. Returns test name and code ref.
+    """
+    (level, verbose, expect) = test_info
+
+    name = 'test_' + logging.getLevelName(level).lower()
+    if verbose:
+        name += '_verbose'
+
+    def _run_test(self):
+        self._test_basic(level, verbose, expect.strip()) # pylint: disable=W0212
+
+    return [name, _run_test]
```

Comment:
Just for future reference, in ``unittest2`` and python3's ``unittest``, there is a ``subtest`` context manager that does this with less code.  It always runs all subtests, and shows the distinguishing features of any failures automatically.  Unf., the documentation is terrible.  It works like this:

```python
for bar in (thing, other_thing, third_thing):
    with unittest2.subtest(foo=bar):  # any kwargs here show up in failure messages, otherwise they're not used.
        result = some_method(foo)  # could just use bar directly if you wanted
        assertWhatever(result)
```

Reply:
(what you have here is fine)